parser_library/src/processing/statement_analyzers/hit_count_analyzer.h

```diff
@@ -15,33 +15,63 @@
 #ifndef PROCESSING_HIT_COUNT_ANALYZER_H
 #define PROCESSING_HIT_COUNT_ANALYZER_H
 
-#include <functional>
-#include <string>
-#include <string_view>
 #include <unordered_map>
+#include <utility>
+#include <vector>
 
-#include "context/hlasm_context.h"
-#include "protocol.h"
+#include "context/id_storage.h"
 #include "statement_analyzer.h"
 #include "utils/general_hashers.h"
 #include "utils/resource_location.h"
 
+namespace hlasm_plugin::parser_library::context {
+class hlasm_context;
+} // namespace hlasm_plugin::parser_library::context
+
 namespace hlasm_plugin::parser_library::processing {
-struct hit_count_details
+struct hit_count
 {
-    range r;
-    size_t count;
+    size_t count = 0;
+    bool contains_statement = false;
+
+    hit_count& operator+=(const hit_count& other_hc)
+    {
+        count += other_hc.count;
+        contains_statement |= other_hc.contains_statement;
+        return *this;
+    }
 };
 
-struct stmt_hit_count_details
+struct hit_count_details
 {
-    std::unordered_map<size_t, hit_count_details> stmt_hc_map;
+    std::vector<hit_count> line_hits;
     bool is_external_macro = true;
+    size_t max_lineno = 0;
+
+    hit_count_details()
+        : line_hits(1000)
+    {}
+
+    static void merge(hit_count_details& to, const hit_count_details& from)
```

Comment:
```suggestion
    hit_count_details& merge(const hit_count_details& from)
```
returning *this. So we can do x.merge(y).merge(z).... if needed.